Archive/Imo/Imo1982Q1.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2024 Alex Brodbelt. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Brodbelt, Violeta Hernández
+-/
+
+import Mathlib.Tactic.NormNum
+import Mathlib.Data.PNat.Basic
+
+/-!
+# IMO 1982 Q1
+
+The function $f(n)$ is defined for all postive integers $n$ and takes on
+non-negative integer values. Also, for all $m$, $n$
+
+$f (m + n) - f(m) - f(n) = 0 \text{ or } 1$
+$f(2) = 0, f(3) > 0, and f(9999) = 3333.$
```

Comment:
```suggestion
The function `f` is defined for all positive integers `n` and takes on
non-negative integer values. Also, for all `m`, `n`

`f (m + n) - f(m) - f(n) = 0` or `1`
`f(2) = 0`, `f(3) > 0`, and `f(9999) = 3333`.
```

Reply:
If we're going to write this using Lean syntax rather than LaTeX syntax, we should probably write `f 2` instead of `f(2)`, etc.

Reply:
I went for writing it in Lean syntax if that's ok with both of you.